File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/model/JcrModifiable.java
package solutions.digamma.damas.jcr.model;

import solutions.digamma.damas.WorkspaceException;
import solutions.digamma.damas.Modifiable;
import solutions.digamma.damas.inspection.NotNull;

import javax.jcr.Property;
import java.util.Calendar;

/**
 * JCR modifiable.
 *
 * @author Ahmad Shahwan
 */
public interface JcrModifiable extends Modifiable, JcrEntity {

    /**
     * Login with which last modification was done.
     *
     * @return
     */
    @Override
    @NotNull
    default String getModifiedBy() throws WorkspaceException {
        return this.getString(Property.JCR_LAST_MODIFIED_BY);
    }

    /**
     * Last modification timestamp.
     *
     * @return
     */
    @Override
    @NotNull
    default Calendar getModificationDate() throws WorkspaceException {
        return this.getDate(Property.JCR_LAST_MODIFIED);
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/model/JcrModifiable.kt
package solutions.digamma.damas.jcr.model

import solutions.digamma.damas.common.WorkspaceException
import solutions.digamma.damas.entity.Modifiable

import javax.jcr.Property

/**
 * JCR modifiable.
 *
 * @author Ahmad Shahwan
 */
internal interface JcrModifiable : Modifiable, JcrEntity {

    /**
     * Login with which last modification was done.
     *
     * @return
     */
    override fun getModifiedBy() =
            this.getString(Property.JCR_LAST_MODIFIED_BY)

    /**
     * Last modification timestamp.
     *
     * @return
     */
    override fun getModificationDate() =
            this.getDate(Property.JCR_LAST_MODIFIED)
}
